### PR TITLE
Crc helpers

### PR DESCRIPTION
**Pull request by michaelcfanning**
Add a CRC algorithm + tests and switch ADO to use it.

@eddynaka

### CODE REVIEW

File: Src/.editorconfig [hunk 1]
```diff
@@ -1,6 +1,15 @@
 # Remove the line below if you want to inherit .editorconfig settings from higher directories
 root = true
 
+# Ignore paths
+[/sarif-pattern-matcher/Src/sarif-sdk/src/sarif/Autogenerated/**]
```
Comment (michaelcfanning) [inline Src/.editorconfig:5]: >matcher [](http://example.com/codeflow?start=16&length=7)

I'm having trouble making sure this folder isn't analyzed in the IDE. Do you know whether this should be working?

File: Src/Plugins/Security/Security.csproj [hunk 2]
```diff
@@ -5,7 +5,7 @@
   <Import Project="$(MSBuildThisFileDirectory)..\..\..\Targets\build.plugins.props" />
 
   <PropertyGroup Label="AssemblyAttributes">
-    <TargetFrameworks>netstandard2.0;net472</TargetFrameworks>
+    <TargetFrameworks>netstandard2.1;net472</TargetFrameworks>
```
Comment (michaelcfanning) [inline Src/Plugins/Security/Security.csproj:8]: >1 [](http://example.com/codeflow?start=35&length=1)

Do you know why this plug is inconsistent with other MS compilation settings?

Comment (eddynaka) [reply Src/Plugins/Security/Security.csproj:8]: we started with 2.0, but the other package got a higher dependency that required 2.1.

Comment (eddynaka) [reply Src/Plugins/Security/Security.csproj:8]: In the security.tests test, can you change to netstandard2.1? (the strings in the unit tests)
that will probably fix the build issue

File: Src/.editorconfig (hunk 1 above)
Comment (eddynaka) [reply Src/.editorconfig:5]: what are you trying to trigger here?
for this to work we would need to change the sarif-sdk and place a file like this in the autogenerated folder.

Comment (michaelcfanning) [reply Src/.editorconfig:5]: The IDE is showing analysis errors for the Autogenerated folder.

File: Src/Plugins/Security/Security.csproj (hunk 2 above)
Comment (eddynaka) [reply Src/Plugins/Security/Security.csproj:8]: fixed.

**Review by eddynaka — APPROVED** [review]
:shipit:

**Pull request merged**